### PR TITLE
Adds warm (hot) (very hot) paper dispenser maintenance MOD

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Adds a paper dispenser mod to the maintenance MOD loot pile.
The paper dispensed has a small chance to burst into flames with every paper made.

image

Inspired by Baystation's rigsuit device, but more literally. 
image

## Why It's Good For The Game

mmmm paper. warm paper. paper on the go. Everything you could ever ask for.

## Changelog

:cl: Melbert
expansion: Paper dispenser maintenance loot MOD.
/:cl:

### CODE REVIEW

File: code/modules/mod/modules/modules_maint.dm [hunk 1]
```diff
@@ -182,6 +182,46 @@
...
+	var/obj/structure/table/nearby_table = locate() in range(1, mod.wearer)
+	to_chat(mod.wearer, span_notice("You dispense a sheet of [crisp_paper.name] [nearby_table ? "onto [nearby_table]":"into your hands"]. It's warm, almost steaming."))
+	playsound(get_turf(src), 'sound/machines/click.ogg', 50, TRUE)
+
+	mod.wearer.put_in_hands(crisp_paper)
+	if(nearby_table)
+		mod.wearer.transferItemToLoc(crisp_paper, nearby_table.drop_location(), silent = FALSE)
+
+	// Up to a 30% chance per sheet to set on fire, 2% per sheet made
+	if(prob(min(num_sheets_dispensed * 2, 30)))
```
Comment (Iamgoofball) [inline code/modules/mod/modules/modules_maint.dm:218]:
```suggestion
	if(prob(min(num_sheets_dispensed * 2, 30)))
		mod.wearer.balloon_alert(mod.wearer, "PC LOAD LETTER")
```

Comment (TemporalOroboros) [reply code/modules/mod/modules/modules_maint.dm:218]: Isn't there literally an error code for this?

File: code/modules/mod/modules/modules_maint.dm [hunk 2]
```diff
@@ -182,6 +182,49 @@
...
+	incompatible_modules = list(/obj/item/mod/module/paper_dispenser)
+	cooldown_time = 5 SECONDS
+	/// The total number of sheets created by this MOD. The more sheets, them more likely they set on fire.
+	var/num_sheets_dispensed = 0
+
+/obj/item/mod/module/paper_dispenser/on_use()
+	. = ..()
+	if(!.)
+		return
+	if(!do_after(mod.wearer, 1 SECONDS))
```
Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:203]: set target to the mod

File: code/modules/mod/modules/modules_maint.dm [hunk 3]
```diff
@@ -182,6 +182,49 @@
 	mod.wearer.put_in_hands(balloon)
 	drain_power(use_power_cost)
 
+/// Paper dispenser - dispenses (sometimes burning) paper sheets
```
Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:185]: nitpick but the other modules have both words capitalized and a dot at the end in their comment

**Comment by itseasytosee** [comment]
Great, more random maint loot standing between me and my precious sticky tape.

**Comment by optimumtact** [comment]
this just creates paper out of thin air? you couldn't even require organic matter to make it work? 

Kinda lazy

**Review by Krysonism — APPROVED** [review]
sprite is good.

**Pull request merged**